src/token/api/src/canister/is20_transactions.rs

```diff
@@ -35,35 +39,36 @@ pub fn transfer_include_fee(
         return Err(TxError::AmountTooSmall);
     }
 
-    if balances.balance_of(&caller.inner()) < amount {
+    if balances.balance_of(from) < amount {
         return Err(TxError::InsufficientBalance);
     }
 
-    charge_fee(balances, caller.inner(), fee_to, fee, fee_ratio)
-        .expect("never fails due to checks above");
+    charge_fee(balances, from, fee_to, fee, fee_ratio).expect("never fails due to checks above");
     transfer_balance(
         balances,
-        caller.inner(),
-        caller.recipient(),
+        from,
+        to,
         (amount - fee).expect("amount > fee is checked above"),
     )
     .expect("never fails due to checks above");
 
-    let id = ledger.transfer(caller.inner(), caller.recipient(), amount, fee);
+    let id = ledger.transfer(from, to, amount, fee);
     Ok(id)
 }
 
 pub fn batch_transfer(
```

Comment:
Change the state deconstruction to: 
```
    let CanisterState {
        ref mut balances,
        ref mut ledger,
        ref bidding_state,
        ref stats,
        ..
    } = &mut *state;
```
so we can call `let id = ledger.batch_transfer(from, transfers, fee);` on the penultimate line of this function